### PR TITLE
Add election date input

### LINKED ISSUE #7: Election Date Input
```
<ElectionDateInput
  defaultValue: 1/1/2021
  onDone=()=>({valid: bool, value: any})
>
```
Date input should be strait text, so someone can copy and paste the data into this field.  Date should be checked for validity. Or find a data picker but it must allow a date to be pasted in.
see react jalaali date picker, or calendar-datepicker

Here is the figma: https://www.figma.com/file/IQKPx02pkBErpmhQoECoq9/Undebate?node-id=151%3A2458
image

And consider this for the date picker from material ui
image

### PR DESCRIPTION
**Pull request by BolunThompson**
Closes #7

### CODE REVIEW

**Comment by BolunThompson** [comment]
Here's the new version with the suggested changes.

**Comment by ddfridley** [comment]
@BolunThompson 
I do see that if I click on the calendar icon, and then click away, that the calendar goes away. 

But - If I click on the calendar icon, and then click on a date, I get this error:
Cannot read properties of undefined (reading 'value')
TypeError: Cannot read properties of undefined (reading 'value')
    at onDatePickerChange (http://localhost:6006/main.58255bcb5f43d8bc5129.bundle.js:1110:25)
    at Calendar.<anonymous> (http://localhost:6006/vendors~main.58255bcb5f43d8bc5129.bundle.js:185198:15)
    at callCallback (http://localhost:6006/vendors~main.58255bcb5f43d8bc5129.bundle.js:206412:12)
    at commitUpdateQueue (http://localhost:6006/vendors~main.58255bcb5f43d8bc5129.bundle.js:206433:9)
    at commitLifeCycles (http://localhost:6006/vendors~main.58255bcb5f43d8bc5129.bundle.js:213780:11)
    at commitLayoutEffects (http://localhost:6006/vendors~main.58255bcb5f43d8bc5129.bundle.js:216725:7)
    at HTMLUnknownElement.callCallback (http://localhost:6006/vendors~main.58255bcb5f43d8bc5129.bundle.js:194110:14)
    at Object.invokeGuardedCallbackDev (http://localhost:6006/vendors~main.58255bcb5f43d8bc5129.bundle.js:194159:16)
    at invokeGuardedCallback (http://localhost:6006/vendors~main.58255bcb5f43d8bc5129.bundle.js:194214:31)
    at commitRootImpl (http://localhost:6006/vendors~main.58255bcb5f43d8bc5129.bundle.js:216463:9)

Also, can you merge with main.  We have taken package-lock.json out of the repo in order to solve the packack-lock can't merge problems for now.

Thanks,
david.

**Comment by ddfridley** [comment]
@BolunThompson this is great! and it meets all the original requirements.  But over the past week in a couple of discussions we decided to change the convention so that components accept the onDone prop which will be called with {valid: bool, value: any}. This way we don't have to have the onChange prop.  I should have told you about it at the time we made the decision.  Can you do this one more change?  Thanks.

**Comment by ddfridley** [comment]
@BolunThompson Here are the conventions - I've been trying to capture it all in the [readme](https://github.com/EnCiv/undebate-ssp#notes-on-react-component-guidelines)  There is another one, number 8 that you should consider also.  Thanks.

File: app/components/election_date_input.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+  const datePickerButtonOnClick = (textDate, datePickerOpen) => {
+    blurDateInput(textDate);
+    setDatePickerOpen(!datePickerOpen);
+  };
+  const blurDateInput = (textDate) => {
+    if (textDate !== "" && !isMdyValid(textDate)) {
+      setError("Please enter a valid date");
+    }
+    propOnDone({valid: error == null, value: mdyToDate(textDate)});
```
Comment (ddfridley) [inline app/components/election_date_input.js:118]: When I enter a date and the blur it works fine.  But when I delete the date, and then blur, onDone gets called with valid true.  Consider that the call to setError does not change the value of error until the next render.   Other than this, it looks great.

**Comment by ddfridley** [comment]
@BolunThompson don't hate me.  If I enter a valid date and then hit tab it shows valid true and the date.  If I click in the field and delete what's there and then hit tab, it still shows valid: true - though the date is empty.  It needs to show valid false in this case.  The parent component needs to know the entry is not valid without examining it.   Does this make sense? thx

**Review by ddfridley — APPROVED** [review]
Works! Looks good. Thanks the the PR.

**Pull request merged**